### PR TITLE
Apicurio hide static openapi document

### LINKED ISSUE #24146: Including quarkus-apicurio-registry-avro pollutes the OpenAPI / Swagger UI with non-working endpoints and data types
### Describe the bug

Including the quarkus-apicurio-registry-avro dependency (for schema registry dev services), pollutes the service's Swagger UI and OpenAPI with non-working (404) endpoints from the Apicurio registry. This happens both in dev mode and when the service is deployed to a non-dev stage.

### Expected behavior

No Apicurio API definitions should be added to the service's API.

### Actual behavior

The service-defined API and Apicurio's non-woking API are presented in the API UI / endpoint.

### How to Reproduce?

1. Include the quarkus-apicurio-registry-avro dependency in the service
2. Go to the Swagger UI or call /openapi

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

17

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.7.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven 3.8.4

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by phillip-kruger**
fix #24146

Also update to 2.1.21 (see https://github.com/smallrye/smallrye-open-api/releases/tag/2.1.21)

### CODE REVIEW

File: extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java [hunk 1]
```diff
@@ -806,27 +821,53 @@ private OpenAPI generateAnnotationModel(IndexView indexView, Capabilities capabi
...
+                if (!shouldIgnore(ignorePatterns, url.toString())) {
+                    // Add as static model
+                    try (InputStream inputStream = url.openStream()) {
+                        if (inputStream != null) {
+                            byte[] contents = IoUtil.readBytes(inputStream);
+
+                            results.add(new Result(format, new ByteArrayInputStream(contents)));
+                        }
+                    } catch (IOException ex) {
+                        ex.printStackTrace();
```
Comment (gastaldi) [inline extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java:851]: Better rethrow this exception (or `Log.error`) 

```suggestion
                        throw new UncheckedIOException(ex);
```

File: extensions/smallrye-openapi/spi/src/main/java/io/quarkus/smallrye/openapi/deployment/spi/IgnoreStaticDocumentBuildItem.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package io.quarkus.smallrye.openapi.deployment.spi;
+
+import java.util.regex.Pattern;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+public final class IgnoreStaticDocumentBuildItem extends MultiBuildItem {
```
Comment (gastaldi) [inline extensions/smallrye-openapi/spi/src/main/java/io/quarkus/smallrye/openapi/deployment/spi/IgnoreStaticDocumentBuildItem.java:7]: Add a meaningful javadoc so this is documented in https://quarkus.io/guides/all-builditems

**Review by gastaldi — CHANGES_REQUESTED** [review]

File: extensions/apicurio-registry-avro/deployment/src/main/java/io/quarkus/apicurio/registry/avro/ApicurioRegistryAvroProcessor.java [hunk 3]
```diff
@@ -64,6 +65,13 @@ void registerSPIClient(BuildProducer<ServiceProviderBuildItem> services) throws
                         "io.apicurio.rest.client.VertxHttpClientProvider"));
     }
 
+    @BuildStep
+    void ignoreIncudedOpenAPIDocument(BuildProducer<IgnoreStaticDocumentBuildItem> ignoreStaticDocumentProducer) {
```
Comment (gastaldi) [inline extensions/apicurio-registry-avro/deployment/src/main/java/io/quarkus/apicurio/registry/avro/ApicurioRegistryAvroProcessor.java:69]: Minor typo
```suggestion
    void ignoreIncludedOpenAPIDocument(BuildProducer<IgnoreStaticDocumentBuildItem> ignoreStaticDocumentProducer) {
```

File: extensions/smallrye-openapi/spi/src/main/java/io/quarkus/smallrye/openapi/deployment/spi/IgnoreStaticDocumentBuildItem.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package io.quarkus.smallrye.openapi.deployment.spi;
+
+import java.util.regex.Pattern;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+/**
+ * Add regex pattern to ignore static OpenAPI document included in extension and/or dependencies
```
Comment (gastaldi) [inline extensions/smallrye-openapi/spi/src/main/java/io/quarkus/smallrye/openapi/deployment/spi/IgnoreStaticDocumentBuildItem.java:8]:
```suggestion
 * Ignore a static OpenAPI document included in extension and/or dependencies. Supports regular expressions.
```

File: extensions/apicurio-registry-avro/deployment/src/main/java/io/quarkus/apicurio/registry/avro/ApicurioRegistryAvroProcessor.java (hunk 3 above)
Comment (phillip-kruger) [reply extensions/apicurio-registry-avro/deployment/src/main/java/io/quarkus/apicurio/registry/avro/ApicurioRegistryAvroProcessor.java:69]: Damm ! thanks fixing now

**Review by gastaldi — APPROVED** [review]

**Review by gsmet — APPROVED** [review]

**Pull request merged**